pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -587,9 +587,26 @@ func (h *appHandler) recodeTrackedResource(resourceName string, resource runtime
 	return nil
 }
 
+type garbageCollectFunc func(ctx context.Context, h *appHandler) error
+
+// 1. collect useless across-namespace resource
+// 2. collect appRevision
+func garbageCollection(ctx context.Context, h *appHandler) error {
+	collectFuncs := []garbageCollectFunc{
+		garbageCollectFunc(gcAcrossNamespaceResource),
+		garbageCollectFunc(cleanUpApplicationRevision),
+	}
+	for _, collectFunc := range collectFuncs {
+		if err := collectFunc(ctx, h); err != nil {
+			return err
```

Comment:
if the GC has error, log it instead of block the main logic

Reply:
after #1463 merged, we can add a event for it if GC errors

Reply:
Fixed